serverless/serverless
# Switch to jest 2

## What did you implement:

This parallelise the the tests in the integration test-suite and lets max 5 workers run at the same time. Right now it take about 17 minutes and we could it get it even down to the longest running tests which is probably about 3-6 minutes if we run all tests in parallel. Unfortunately CloudFormation has a rate limit. While we could ask AWS to increase I recommend we stay with the standard setup. Otherwise newcomers to the repo won't be able to run it out of the box and they need to understand the rate limit and how to manipulate the max amount of workers.

## How did you implement it:

It's based on @dougmoscrop PR https://github.com/serverless/serverless/pull/2654 which made it possible to parallelise the test suite and in addition makes sure everything is cleaned up properly. 

I used jest codemod to convert the tests with this codemod and then did some manual cleanup.

## How can we verify it:

```bash
npm run complex-integration-test
```

As you can see I also ran the test suite on Travis and then deactivated it again. There was a previous PR, where some tests failed. Locally and on this PR it succeeded. Maybe a caching issue, but we will see in the future.

## Future plans

After this being merged I want to migrate the unit-test suite as well. Due the similarity with mocha there won't be many changes. We could also switch from chai to jasmine2 at some point, but I believe that's not urgent.

## Todos:



***Is this ready for review?:*** YES
